compiler/rustc_middle/src/ty/list.rs

```diff
@@ -200,7 +200,10 @@ unsafe impl<T: Sync> Sync for List<T> {}
 
 // We need this since `List` uses extern type `OpaqueListContents`
 #[cfg(parallel_compiler)]
-unsafe impl<T: Sync> rustc_data_structures::sync::DynSync for List<T> {}
+unsafe impl<T: rustc_data_structures::sync::DynSync> rustc_data_structures::sync::DynSync
```

Comment:
Probably should just import `DynSync` here.

Reply:
Done :)